galasa-managers-parent/galasa-managers-core-parent/dev.galasa.textscan.manager/src/main/java/dev/galasa/textscan/ILogScanner.java

```diff
@@ -0,0 +1,314 @@
+     * 
+     * @param searchString - The text to search for
+     * @param failString - Failure text to search for, can be null meaning no fail search
+     * @param count - Atleast how many occurrences of the searchString must exist
+     * @return This log scanner for fluent calls
+     * @throws FailTextFoundException - If the failString was found
+     * @throws MissingTextException - If no occurrences of the searchString was found 
+     * @throws IncorrectOccurancesException - If insufficient occurrences were found, if there are zero occurrences, then MissingTextException will be thrown
+     */
+    ILogScanner scanSinceCheckpoint(String searchString, String failString, int count) throws MissingTextException, IncorrectOccurancesException;
+    
+    /**
+     * Convenience method for scanSinceCheckpoint(searchString, failString, 1)
+     * 
+     * @param searchString - The text to search for
+     * @param failString - Failure text to search for, can be null meaning no fail search
+     * @return This log scanner for fluent calls
+     * @throws FailTextFoundException - If the failString was found
+     * @throws MissingTextException - If no occurrences of the searchString was found 
+     */
+    ILogScanner scanSinceCheckpoint(String searchString, String failString) throws FailTextFoundException, MissingTextException;
+    
+    /**
+     * Convenience method for scanSinceCheckpoint(searchString, null, 1)
+     * 
+     * @param searchString - The text to search for
+     * @return This log scanner for fluent calls
+     * @throws FailTextFoundException - If the failString was found
+     */
+    ILogScanner scanSinceCheckpoint(String searchString) throws MissingTextException;
+    
+    
+    
+    /**
+     * Will search the log from the checkpoint looking for the searchPattern.  When it finds the "occurrence" of the string, it will return the found text.
+     * Useful for returning the actual value of the searchPattern
+     * 
+     * @param searchPattern - The searchPattern to look for
+     * @param occurrence - The occurrence to be returned
+     * @return The text of the searchPattern found
+     * @throws MissingTextException - The searchPattern was not found at all
+     * @throws IncorrectOccurancesException - The searchPattern was found, but not the x occurrence
+     */
+    String scanForMatchSinceCheckpoint(Pattern searchPattern, int occurrance) throws MissingTextException, IncorrectOccurancesException;
+    
+    /**
+     * Convenience method for scanForMatchSinceCheckpoint(searchPattern, 1)
+     * 
+     * @param searchPattern - The searchPattern to look for
+     * @return The text of the searchPattern found
+     * @throws MissingTextException - The searchPattern was not found at all
+     */
+    String scanForMatchSinceCheckpoint(Pattern searchPattern) throws MissingTextException;
+    
+    /**
+     * Convenience method for scanForMatchSinceCheckpoint(Pattern.Compile("\Q" + searchString + "\E"), occurrence)
```

Comment:
I'm not a fan of overloading the signatures with convenience methods.  Why not make the convenience methods the default and remove the signatures that contain the Pattern arguments?  Unless there is an oblique performance or functional argument to be made I think this considerably cleans up the interface and makes it more consumable

Reply:
However I am a fan of overloaded methods because it cleans up the test code if there are a lot of parameters that are not using,  eg. scan(text, "search", null, 1) is ugly compared to scan(text, "search")
It only requires ONE line of code in the manager.  I have had complaints from users of JAT about the necessity having to provide default parameters on worse methods,   so if if we do it for some, we should do them for all.

Reply:
Yes in those cases I agree.  But why have a set of methods with a Pattern and the same set replacing the Pattern for a String.  Can't we just default to the String convenience?

Reply:
I think I prefer the overloading here. It doesnt confuse the method purpose and it does clean up and simplify using the methods inside tests

Reply:
I have removed the overloading temporarily.   Think we need a team discussion on if we have overloading or not and standardise.